From b1235ae2a00eb18fc85e630c1de5e8df39a2afd2 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 16 Nov 2011 14:06:52 +0800
Subject: [PATCH 15/18] fs: pramfs: Remove unnecesary while loop in pram_xattr_get

Upstream ID: 7b6ae67a22a116a20f84a10531b41a627f8db31d
  git://pramfs.git.sourceforge.net/gitroot/pramfs/pramfs
  From: Marco <marco@ncc-1701-f.site>

Integrated-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 fs/pramfs/xattr.c |   10 +---------
 1 files changed, 1 insertions(+), 9 deletions(-)

diff --git a/fs/pramfs/xattr.c b/fs/pramfs/xattr.c
index 351f1b0..860d334 100644
--- a/fs/pramfs/xattr.c
+++ b/fs/pramfs/xattr.c
@@ -217,14 +217,6 @@ bad_block:	pram_err(sb, "inode %ld: bad block %llu", inode->i_ino,
 			goto found;
 		entry = next;
 	}
-	/* Check the remaining name entries */
-	while (!IS_LAST_ENTRY(entry)) {
-		struct pram_xattr_entry *next =
-			PRAM_XATTR_NEXT(entry);
-		if ((char *)next >= end)
-			goto bad_block;
-		entry = next;
-	}
 
 	desc = GET_DESC(sbi, blocknr);
 	if (IS_ERR(desc)) {
@@ -382,7 +374,7 @@ ssize_t pram_listxattr(struct dentry *dentry, char *buffer, size_t size)
 /*
  * pram_xattr_set()
  *
- * Create, replace or remove an extended attribute for this inode. Buffer
+ * Create, replace or remove an extended attribute for this inode. Value
  * is NULL to remove an existing extended attribute, and non-NULL to
  * either replace an existing extended attribute, or create a new extended
  * attribute. The flags XATTR_REPLACE and XATTR_CREATE
-- 
1.7.0.4

